lib/Helper/Translate.php

```diff
@@ -120,6 +120,55 @@ public static function InitLocale($config, $language = NULL)
         self::$jsLocaleRequested = str_replace('_', '-', self::$requestedLanguage);
     }
 
+    /**
+     * Get translations for user selected language
+     * @param $config
+     * @param $language
+     * @return Translator
+     */
+    public static function getTranslationsFromLocale($config, $language = NULL): Translator
+    {
+        // The default language
+        $default = ($language === null) ? $config->getSetting('DEFAULT_LANGUAGE') : $language;
+
+        // Build an array of supported languages
+        $localeDir = PROJECT_ROOT . '/locale';
+        $supportedLanguages = array_map('basename', glob($localeDir . '/*.mo'));
+
+        // Record any matching languages we find.
+        $foundLanguage = null;
+
+        // Try to get the local firstly from _REQUEST (post then get)
+        if ($language != null) {
+            // Serve only the requested language
+            // Firstly, Sanitize it
+            self::$requestedLanguage = str_replace('-', '_', $language);
+
+            // Check its valid
+            if (in_array(self::$requestedLanguage . '.mo', $supportedLanguages)) {
+                $foundLanguage = self::$requestedLanguage;
+            }
+        }
+
+        // Are we still empty, then default language from settings
+        if ($foundLanguage == '') {
```

Comment:
Do we want to set a default? Perhaps it would be better to return NULL if that language isn't supported, so that we can handle it in the calling code (which you have already I think).